### ISSUE TITLE
No GUI logo file found

### ISSUE THREAD

**Issue by jlstevens** [ContinuumIO/intake#230]
When running:

```
import intake
cat = intake.open_catalog('../catalog.yml')
```

I get the following error about a logo file not being found somewhere inside ipywidgets:

<details>
---------------------------------------------------------------------------
FileNotFoundError                         Traceback (most recent call last)
<ipython-input-1-283997e3fbf0> in <module>
----> 1 import intake
      2 cat = intake.open_catalog('../catalog.yml')

~/miniconda3/envs/analytics/lib/python3.7/site-packages/intake/__init__.py in <module>
     12 from .source import registry
     13 from .source.discovery import autodiscover
---> 14 from .gui import DataBrowser
     15 
     16 # Populate list of autodetected plugins

~/miniconda3/envs/analytics/lib/python3.7/site-packages/intake/gui/__init__.py in <module>
      1 try:
      2     import ipywidgets
----> 3     from .widgets import DataBrowser
      4 
      5 except ImportError:

~/miniconda3/envs/analytics/lib/python3.7/site-packages/intake/gui/widgets.py in <module>
      8 here = os.path.abspath(os.path.dirname(__file__))
      9 logo_file = os.path.join(here, 'logo.png')
---> 10 logo = widgets.Box([widgets.Image.from_file(logo_file)],
     11                    layout=widgets.Layout(width='80px', height='80px'))
     12 

~/miniconda3/envs/analytics/lib/python3.7/site-packages/ipywidgets/widgets/widget_media.py in from_file(cls, filename, **kwargs)
    165     @classmethod
    166     def from_file(cls, filename, **kwargs):
--> 167         return cls._from_file('image', filename, **kwargs)
    168 
    169     def __repr__(self):

~/miniconda3/envs/analytics/lib/python3.7/site-packages/ipywidgets/widgets/widget_media.py in _from_file(cls, tag, filename, **kwargs)
     51         Returns an `Media` with the value set from the filename.
     52         """
---> 53         value = cls._load_file_value(filename)
     54 
     55         if 'format' not in kwargs:

~/miniconda3/envs/analytics/lib/python3.7/site-packages/ipywidgets/widgets/widget_media.py in _load_file_value(cls, filename)
    102             return filename.read()
    103         else:
--> 104             with open(filename, 'rb') as f:
    105                 return f.read()
    106 

FileNotFoundError: [Errno 2] No such file or directory: '/Users/jstevens/miniconda3/envs/analytics/lib/python3.7/site-packages/intake/gui/logo.png'
</details>

As no GUI was requested or mentioned in the code, I would not expect to see such an error. In general, I would hope that anything relating to ipywidgets would not prevent intake being used unless GUI/widget elements are explicitly requested.

This is using:

```
conda install -c intake/label/dev intake intake-xarray
```

**Comment by jbednar** [comment]
I had thought this general issue was already addressed; i.e. that no gui code was being run when the gui wasn't requested.  Maybe that fix is on master but not yet released?

**Comment by martindurant** [comment]
We are no longer publishing dev builds, apparently - I thought we were (to be checked). As you specifically select it, you are getting a version (0.2.6) that is a few versions older than a main release. Both conda-forge and intake channels have 0.3.0

The relevant PR https://github.com/ContinuumIO/intake/pull/206

**Comment by seibert** [comment]
If I understand the Github branch list shown on the commit, I don't think #206 has appeared in a tagged release yet.

**Comment by martindurant** [comment]
Actually, you may be right. @jbednar , you would like a release?

**Comment by jlstevens** [comment]
Ah ok, there is no issue using:

```
conda install -c intake intake intake-xarray
```

In which case the latest releases should be labelled `dev` - I would always expect asking for `label/dev` to get me the latest release.

**Comment by jbednar** [comment]
I agree with the principle that the dev channel should never trail main; it should always have the bleeding edge, even if that's a real release.

Separately, it would be good to have at least a dev release that avoids running the gui unless requested, yes. Thanks!

**Comment by seibert** [comment]
You should be able to do that.  You can make your upload step detect which branch it is running under and only upload in that case.  I think Travis CI sets some environment variables to enable this.

**Comment by jlstevens** [comment]
> I uploaded a hand-made dev-label package just for you

Thanks! Unfortunately, this isn't what I ended up using. The version of intake/intake-xarray that worked for me was found on conda-forge.

Does that mean I should just use conda-forge in future and forget about the intake channel?

**Comment by martindurant** [comment]
OK to close, @jlstevens ?

**Comment by jlstevens** [comment]
@martindurant  Sure!
